sdl_android/src/main/java/com/smartdevicelink/protocol/SdlProtocol.java

```diff
@@ -0,0 +1,1457 @@
+            if(serviceType.equals(SessionType.RPC)){
+                hashID = (Integer) packet.getTag(ControlFrameTags.RPC.StartServiceACK.HASH_ID);
+                Object version = packet.getTag(ControlFrameTags.RPC.StartServiceACK.PROTOCOL_VERSION);
+
+                if(version!=null) {
+                    //At this point we have confirmed the negotiated version between the module and the proxy
+                    protocolVersion = new Version((String) version);
+                }else{
+                    protocolVersion = new Version("5.0.0");
+                }
+
+                //Check to make sure this is a transport we are willing to accept
+                TransportRecord transportRecord = packet.getTransportRecord();
+
+                if(transportRecord == null || !requestedPrimaryTransports.contains(transportRecord.getType())){
+                    onTransportNotAccepted("Transport is not in requested primary transports");
+                    return;
+                }
+
+
+                // This enables custom behavior based on protocol version specifics
+                if (protocolVersion.isNewerThan(new Version("5.1.0")) >= 0) {
+
+                    if (activeTransports.get(SessionType.RPC) == null) {    //Might be a better way to handle this
+
+                        ArrayList<String> secondary = (ArrayList<String>) packet.getTag(ControlFrameTags.RPC.StartServiceACK.SECONDARY_TRANSPORTS);
+                        ArrayList<Integer> audio = (ArrayList<Integer>) packet.getTag(ControlFrameTags.RPC.StartServiceACK.AUDIO_SERVICE_TRANSPORTS);
+                        ArrayList<Integer> video = (ArrayList<Integer>) packet.getTag(ControlFrameTags.RPC.StartServiceACK.VIDEO_SERVICE_TRANSPORTS);
+
+                        activeTransports.put(SessionType.RPC, transportRecord);
+                        activeTransports.put(SessionType.BULK_DATA, transportRecord);
+                        activeTransports.put(SessionType.CONTROL, transportRecord);
+
+                        //Build out the supported secondary transports received from the
+                        // RPC start service ACK.
+                        supportedSecondaryTransports = new ArrayList<>();
+                        if (secondary == null) {
+                            // If no secondary transports were attached we should assume
+                            // the Video and Audio services can be used on primary
+                            if (requiresHighBandwidth
+                                    && TransportType.BLUETOOTH.equals(transportRecord.getType())) {
+                                //transport can't support high bandwidth
+                                onTransportNotAccepted(transportRecord.getType() + " can't support high bandwidth requirement, and secondary transport not supported.");
+                                return;
+                            }
+
+                            activeTransports.put(SessionType.NAV, transportRecord);
```

Comment:
Probably I am missing something here. But shouldn't we check if audio and video lists contain the primary transport to make sure that the HU supports media before adding the transports to the active transports?

Reply:
Might be a good idea, but check to see if the entry exists as well. Because if it isn't included we have to assume the primary transport is acceptable.

Reply:
Does that look good:
```
if (video == null || video.contains(PRIMARY_TRANSPORT_ID)) {
    activeTransports.put(SessionType.NAV, transportRecord);
}
if (audio == null || audio.contains(PRIMARY_TRANSPORT_ID)) {
    activeTransports.put(SessionType.PCM, transportRecord);
}
```